src/p_spec.c

```diff
@@ -2638,6 +2641,31 @@ void T_Scroll(scroll_t *s)
     }
 }
 
+// [crispy] smooth texture scrolling
+void R_InterpolateTextureOffsets (void)
+{
+  thinker_t *th;
+
+  if (!uncapped || leveltime <= oldleveltime)
+    return;
+
+  for (th = thinkercap.next; th != &thinkercap; th = th->next)
```

Comment:
It's a bit lame to iterate through all thinkers during each rendered frame. Any other ideas?

Reply:
I haven't gone far enough in this aspect of pr+ to completely understand it, but I believe it generates a list of things that need to be interpolated and iterates over that list each frame.

I'm not sure how it builds that list exactly. You could for instance start each gametic with an empty list and add each scroll thinker to it while going through them. Then you could loop over that list here, and empty it when you reach the next gametic. Or maybe you don't need to rebuild the list if you do it cleverly 😄

Haven't thought enough about it, just some ideas.

Reply:
You can have a texture that scrolls based on floor height, so it can start and stop scrolling. I don't know if that applies here or not.

Reply:
> In the first kind, displacement scrolling, the position of the scrolled objects or walls changes proportionally to the motion of the floor or ceiling in the sector on the first sidedef of the scrolling trigger. The proportionality is set by the length of the linedef trigger. If it is 32 units long, the wall, floor, ceiling, or object moves 1 unit for each unit the floor or ceiling of the controlling sector moves. If it is 64 long, they move 2 units per unit of relative floor/ceiling motion in the controlling sector and so on.

From the boom spec

Reply:
It seems PrBoom+ builds the list in `P_RunThinkers` function: https://github.com/coelckers/prboom-plus/blob/60ca146f5a1a9abe15f9c275c3bbe0189c500d4e/prboom2/src/p_tick.c#L250

Reply:
> You can have a texture that scrolls based on floor height, so it can start and stop scrolling. I don't know if that applies here or not.

I see, thanks. But in that case the `dx` and `dy` increments would be `0` and we could still at least calculate the interpolation. I guess that's a lower price to pay than recreating the list during each tic.